### ISSUE TITLE
AWS::ECS::TaskDefinition ContainerDefinitions - False Positive `E3042: At least one essential container is required` when no `Essential` is configured

### ISSUE THREAD

**Issue by r-heimann** [aws-cloudformation/cfn-lint#4535]
### CloudFormation Lint Version

cfn-lint 1.51.3

### What operating system are you using?

Windows 11

### Describe the bug

After updating to cfn-lint 1.51.3 we receive a false-positive error for AWS::ECS::TaskDefinition `ContainerDefinitions`:

```
[cfn-lint] E3042: At least one essential container is required
```

This happens when **not** specifying `Essential: true` in `ContainerDefinitions`:

https://docs.aws.amazon.com/AWSCloudFormation/latest/TemplateReference/aws-properties-ecs-taskdefinition-containerdefinition.html#cfn-ecs-taskdefinition-containerdefinition-essential

> Essential
> If the essential parameter of a container is marked as true, and that container fails or stops for any reason, all other containers that are part of the task are stopped. If the essential parameter of a container is marked as false, its failure doesn't affect the rest of the containers in a task. If this parameter is omitted, a container is assumed to be essential.
> 
> All tasks must have at least one essential container. If you have an application that's composed of multiple containers, group containers that are used for a common purpose into components, and separate the different components into multiple task definitions. For more information, see [Application Architecture](https://docs.aws.amazon.com/AmazonECS/latest/developerguide/application_architecture.html) in the Amazon Elastic Container Service Developer Guide.
> 
> Required: No
> 
> Type: Boolean
> 
> Update requires: [Replacement](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/using-cfn-updating-stacks-update-behaviors.html#update-replacement)

### Expected behavior

cfn-lint doesn't produce a false-positive.

### Reproduction template

WIP
